src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,83 @@
+package duke.task;
+
+import duke.common.CustomException;
+
+import java.util.ArrayList;
+
+/**
+ * Represents all of the users tasks in a list.
+ */
+public class TaskList {
+    private static ArrayList<Task> tasks;
+
+    public TaskList(ArrayList<Task> tasks) {
+        TaskList.tasks = tasks;
+    }
+
+    /**
+     * Adds a task to the task list.
+     * @param task The task to be added
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes a task to the task list.
+     * @param taskindex The index of the task to be deleted in the task list.
+     */
+    public void deleteTask(int taskindex) {
+        tasks.remove(taskindex);
+    }
+
+    /**
+     * Marks a task in the task list as done
+     * @param taskindex The index of the task in the task list that is to be marked as done.
+     */
+    public void markAsDone(int taskindex) {
+        tasks.get(taskindex).markAsDone();
+    }
+
+    /**
+     * Returns current task list.
+     * @return current task list.
+     */
+    public ArrayList<Task> getList() {
+        return tasks;
+    }
+
+    /**
```

Comment:
Some Javadoc format issue, there should be an empty line between the description and your @tag